### ISSUE TITLE
Generate additive tokens with rgb value outputs

### ISSUE THREAD

**Issue by jkinley** [lukasoppermann/style-dictionary-utils#30]
HI Lukas. I have a question on how we might go about doing something and think it could be of value to your repo. 

So (sadly) we are supporting a Bootstrap theme and need some of our tokens to line-up with Bootstrap CSS Vars. One of the things they do is create CSS vars like this:

```
  --bs-primary: #0d6efd;
  --bs-secondary: #6c757d;
  --bs-success: #198754;
  --bs-info: #0dcaf0;
  --bs-warning: #ffc107;
  --bs-danger: #dc3545;
```

But because the hex values don't handle opacity, they also generate RGB values from these hex values, like so:

```
 --bs-primary-rgb: 13, 110, 253;
  --bs-secondary-rgb: 108, 117, 125;
  --bs-success-rgb: 25, 135, 84;
  --bs-info-rgb: 13, 202, 240;
  --bs-warning-rgb: 255, 193, 7;
  --bs-danger-rgb: 220, 53, 69;
```

So how do we go about handling this with Style Dictionary? My colleague Doug has worked out a custom transformer, but we still have some questions. 

1. We only need this for a handful of the color tokens. We don't want to generate these RGB values for 100's of tokens. More like a dozen or so, but it would be nice to automate to reduce maintenance  and errors.

2. Is there a way to 'handpick' or label specific tokens to be processed so that we retain the hex token but also output the RGB values for it as well?

3. Given that the W3C spec wants colors defined in Hex and they suggest conversion to other color spaces, how do you think is the best way to approach this?

<img width="837" alt="image" src="https://github.com/lukasoppermann/style-dictionary-utils/assets/19417886/95ea31b0-a824-4974-abe3-37b60f6f6940">

**Comment by lukasoppermann** [comment]
Hey @jkinley,

firstly, hex values do work with opacity, the format is called `hex8` and the last 2 digits are the [opacity in hex](https://gist.github.com/lopspower/03fb1cc0ac9f32ef38f4).

1. You need to create a filter that can detect which colors need this. This could be as simple as adding a config file or an array with the token names in the filter or you could add a property like `usesOpacity`.

2. You can't output two tokens from one using a transformer. For this you need to create a custom format which can handle this.

3. I think the optimal way is to use `hex8`. If bootstrap needs the rgb values and hex values I would just create a format (this can be the filter and format in one, if this is the only place where you need to filter).

Hope that helps.

**Comment by jkinley** [comment]
> firstly, hex values do work with opacity, the format is called `hex8` and the last 2 digits are the [opacity in hex](https://gist.github.com/lopspower/03fb1cc0ac9f32ef38f4).

Yes. I am aware of this, but it does not help use get the RGB values which we need for Bootstrap. We will look into your other suggestions. Thanks!

**Comment by anoopsinghbayes** [comment]
i have done this with some custom functions (transformers etc)
i feel this will be temporary requirement as modern browser support [relative color](https://caniuse.com/?search=relative%20color)

**Comment by lukasoppermann** [comment]
Can you share more about how you did this? Maybe a PR of we feel it would solve the issue?

**Comment by anoopsinghbayes** [comment]
i will put a minimal example here soon

**Comment by anoopsinghbayes** [comment]
> Can you share more about how you did this? Maybe a PR if we feel it would solve the issue?


i will put a minimal example here soon

**Comment by anoopsinghbayes** [comment]
@lukasoppermann here is a minimal [gist](https://gist.github.com/anoopsinghbayes/c1b9003dc0700cd119f345fb997d23ed)  to showcase this use case

**Comment by lukasoppermann** [comment]
Nice approach @anoopsinghbayes.

I think this is a great solution in general. 
Do you not add a `type: color` and this is why it is not processed by style dictionary afterwards?

**Comment by lukasoppermann** [comment]
> Can you explain what does this mean?
> i see type:color on all the token definitions in my example

Yes, but not on the rgb tokens. It seems like the preprocess: https://gist.github.com/anoopsinghbayes/c1b9003dc0700cd119f345fb997d23ed#file-index-mjs-L5-L68 does not add a type and thus the tokens are ignored in any transformation, correct?
